File: CSharp/LogFunction.cs
//Class to write error / information to a log file

public static string dir = $@"";
public static string path = Path.Combine(dir, "log.txt");

//Log Error
public void Log(string err)
{
    StreamWriter sw = null;
    try {
        if (!Directory.Exists(dir)) {
            Directory.CreateDirectory(dir);
        }
        sw = File.AppendText(path);
        sw.WriteLine($"{DateTime.Now.ToString("dd-MM-yyyy HH:mm:ss")} :: {err}");
    }
    catch (Exception) {
        throw;
    }   
    finally {
        sw.Close();
    }
}

File: Visual Basic/LogFunction.vb
Public Class Log

    Public Shared Sub Write(Byval msg As String)
        Dim path As String = $"{Application.StartupPath}\log.txt"
        Dim sw As StreamWriter = Nothing

        Try
            sw = File.AppendText(path)
            sw.WriteLine($"{Date.Now:dd-MM-yy HH:mm:ss} :: {msg}")
        Catch Ex as Exception
            'exception
        Finally
            sw.Close()
        End Try
    End Sub

End Class